.github/workflows/main.yml

```diff
@@ -96,12 +98,31 @@ jobs:
         if: contains(matrix.tests, ':stubs:')
         run: MOCHA_WEBDRIVER_HEADLESS=1 yarn run test:stubs
 
+      - name: Run gen-server tests with minio and redis
+        if: contains(matrix.tests, ':gen-server-')
+        run: |
+          PGPASSWORD=db_password psql -h localhost -U db_user -w db_name -c "SHOW ALL;" | grep ' jit '
+          export TEST_SPLITS=$(echo $TESTS | sed "s/.*:gen-server-\([^:]*\).*/\1/")
+          yarn run test:gen-server
+          TYPEORM_TYPE=postgres TYPEORM_HOST=localhost TYPEORM_DATABASE=db_name TYPEORM_USERNAME=db_user TYPEORM_PASSWORD=db_password yarn run test:gen-server
+        env:
+          MOCHA_WEBDRIVER_HEADLESS: 1
+          TESTS: ${{ matrix.tests }}
+          GRIST_DOCS_MINIO_ACCESS_KEY: administrator
+          GRIST_DOCS_MINIO_SECRET_KEY: administrator
+          TEST_REDIS_URL: "redis://localhost/11"
+          GRIST_DOCS_MINIO_USE_SSL: 0
+          GRIST_DOCS_MINIO_ENDPOINT: localhost
+          GRIST_DOCS_MINIO_PORT: 9000
+          GRIST_DOCS_MINIO_BUCKET: grist-docs-test
+
```

Comment:
I also think it logically makes more sense to split them, but I put them together to avoid excessive duplication in the `env` section. Do you feel strongly about it?

Reply:
Would yaml's anchor / alias solve your issue? This resource help understanding the concept: https://www.educative.io/blog/advanced-yaml-syntax-cheatsheet

```yaml

      - name: Run gen-server tests with minio and redis with sqlite
        if: contains(matrix.tests, ':gen-server-')
        run: |
          export TEST_SPLITS=$(echo $TESTS | sed "s/.*:gen-server-\([^:]*\).*/\1/")
          yarn run test:gen-server
        env: &gen-server-env-vars
          MOCHA_WEBDRIVER_HEADLESS: 1
          TESTS: ${{ matrix.tests }}
          GRIST_DOCS_MINIO_ACCESS_KEY: administrator
          GRIST_DOCS_MINIO_SECRET_KEY: administrator
          TEST_REDIS_URL: "redis://localhost/11"
          GRIST_DOCS_MINIO_USE_SSL: 0
          GRIST_DOCS_MINIO_ENDPOINT: localhost
          GRIST_DOCS_MINIO_PORT: 9000
          GRIST_DOCS_MINIO_BUCKET: grist-docs-test

      - name: Run gen-server tests with minio and redis with postgresql
        if: contains(matrix.tests, ':gen-server-')
        run: |
          PGPASSWORD=db_password psql -h localhost -U db_user -w db_name -c "SHOW ALL;" | grep ' jit '
          export TEST_SPLITS=$(echo $TESTS | sed "s/.*:gen-server-\([^:]*\).*/\1/")
          yarn run test:gen-server
        env: 
          <<: *gen-server-env-vars
          TYPEORM_TYPE: postgres
          TYPEORM_HOST: localhost
          TYPEORM_DATABASE: db_name
          TYPEORM_USERNAME: db_user
          TYPEORM_PASSWORD: db_password
```

And you might want to factorize a bit more… or not… Sometimes we have to choose a tradeoff between readability and factorization.

Reply:
@fflorent Sadly anchors are not supported by GH actions: https://github.com/actions/runner/issues/1182